src/prefect/tasks/kubernetes/job.py

```diff
@@ -695,34 +698,68 @@ def run(
                 "The job name must be defined in the body under the metadata key."
             )
 
-        api_client = cast(
+        api_client_job = cast(
             client.BatchV1Api, get_kubernetes_client("job", kubernetes_api_key_secret)
         )
 
-        api_client.create_namespaced_job(namespace=namespace, body=body, **kube_kwargs)
+        api_client_pod = cast(
+            client.CoreV1Api, get_kubernetes_client("pod", kubernetes_api_key_secret)
+        )
+
+        api_client_job.create_namespaced_job(
+            namespace=namespace, body=body, **kube_kwargs
+        )
         self.logger.info(f"Job {job_name} has been created.")
 
-        completed = False
-        while not completed:
-            res = api_client.read_namespaced_job_status(
-                name=job_name, namespace=namespace
+        job = api_client_job.read_namespaced_job_status(
+            name=job_name, namespace=namespace
+        )
+
+        pod_selector = f"controller-uid={job.metadata.labels['controller-uid']}"
+        pods_list = api_client_pod.list_namespaced_pod(
+            namespace=namespace, label_selector=pod_selector
+        )
+
+        if len(pods_list.items) != 1:
```

Comment:
Some jobs might run multiple pods

Reply:
Hmm, I was thinking about this, but I thought it was impossible, how can a job create multiple pods?

Reply:
By specifying `parallelism` or `completions`: https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs